File: aspnet/GrabzIt/GrabzItRequest.cs
using GrabzIt.Parameters;
using System;
using System.Collections.Generic;
using System.Text;

namespace GrabzIt
{
    internal class GrabzItRequest
    {
        private BaseOptions instance_options;
        private bool instance_post;
        private string instance_ws_url;
        private string instance_data;

        internal GrabzItRequest()
        {
        }

        internal void Store(string wsUrl, bool post, BaseOptions options)
        {
            Store(wsUrl, post, options, null);
        }

        internal void Store(string wsUrl, bool post, BaseOptions options, string data)
        {
            instance_ws_url = wsUrl;
            instance_post = post;
            instance_options = options;
            instance_data = data;
        }

        internal BaseOptions Options
        {
            get
            {
                return instance_options;
            }
            set
            {
                instance_options = value;
            }
        }

        internal bool IsPost
        {
            get
            {
                return instance_post;
            }
        }

        internal string Data
        {
            get
            {
                return instance_data;
            }
        }

        internal string WebServiceURL
        {
            get
            {
                return instance_ws_url;
            }
        }

	    internal string TargetUrl
	    {
            get
            {
                if (IsPost)
                {
                    return string.Empty;
                }
                return Data;
            }
	    }
    }
}

File: php/lib/GrabzItRequest.php
<?php
namespace GrabzIt;

class GrabzItRequest
{
	private $url;
	private $isPost = false;
	private $options;
	private $data;

	public function __construct($url, $isPost, $options, $data = null)
	{
		$this->url = $url;
		$this->data = $data;
		$this->isPost = $isPost;
		$this->options = $options;
	}

	public function isPost()
	{
		return $this->isPost;
	}
	
	public function getTargetUrl()
	{
		if ($this->isPost)
		{
			return '';
		}
		return $this->data;
	}

	public function getUrl()
	{
		return $this->url;
	}
	
	public function getData()
	{
		return $this->data;
	}

	public function getOptions()
	{
		return $this->options;
	}
}
?>